File: general_customization/public/js/Sales_Invoice.js
frappe.ui.form.on('Sales Invoice', {
    customer: function(frm) {
        if (frm.doc.customer) {
            frappe.call({
                method: "erpnext.accounts.utils.get_balance_on",  
                args: {
                    party_type: 'Customer',  
                    party: frm.doc.customer  
                },

                callback: function(r) {
                   if (r.message) {
                    console.log(r.message);
                        frm.set_value('party_balance', r.message);
                        console.log("Customer selected: " + frm.doc.customer + ", custom balance set: " + r.message);
                    }
                }
                
            });
        }
    }
});

File: general_customization/overrides/sales_invoice.py
from erpnext.accounts.doctype.sales_invoice.sales_invoice import SalesInvoice
from erpnext.accounts.utils import get_balance_on


class CustomSalesInvoice(SalesInvoice):
    def before_save(self):
        self.party_balance = get_balance_on(
            party_type="Customer", party=self.customer_name
        )
